lib/config/configuration.go

```diff
@@ -1006,20 +1006,42 @@ func applyDatabasesConfig(fc *FileConfig, cfg *service.Config) error {
 		}
 		var caBytes []byte
 		var err error
-		if database.CACertFile != "" {
-			caBytes, err = ioutil.ReadFile(database.CACertFile)
+		if database.TLS.CACertFile != "" {
+			caBytes, err = os.ReadFile(database.TLS.CACertFile)
 			if err != nil {
 				return trace.Wrap(err)
 			}
 		}
+
+		// ca_cert_file is deprecated, but we still support it.
+		//Print a warning the old field is still being used.
+		if database.CACertFile != "" {
+			if database.TLS.CACertFile != "" {
+				// New and old fields are set. Ignore the old field.
+				log.Warn("Ignoring deprecated ca_cert_file; using tls.ca_cert_file.")
+			} else {
+				// Only old field is set, inform about deprecation.
+				log.Warn("ca_cert_file is deprecated, please use tls.ca_cert_file instead.")
+
+				caBytes, err = os.ReadFile(database.CACertFile)
+				if err != nil {
+					return trace.Wrap(err)
+				}
+			}
+		}
```

Comment:
Can we move this logic to a separate function ?

Reply:
Sure